<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MRAT â€¢ MRAT</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MRAT">
<meta property="og:description" content="MRAT">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MRAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MRAT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MRAT</h1>
            
      
      
      <div class="hidden name"><code>MRAT.Rmd</code></div>

    </div>

    
    
<p>Install the package</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">##devtools::install_github("ShihChingYu/MRAT")</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">MRAT</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required package: EnsDb.Hsapiens.v75</code></pre>
<pre><code>## Loading required package: ensembldb</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: 'IRanges'</code></pre>
<pre><code>## The following object is masked from 'package:grDevices':
## 
##     windows</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: GenomicFeatures</code></pre>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## Loading required package: AnnotationFilter</code></pre>
<pre><code>## 
## Attaching package: 'ensembldb'</code></pre>
<pre><code>## The following object is masked from 'package:stats':
## 
##     filter</code></pre>
<pre><code>## Loading required package: biomaRt</code></pre>
<p>Load the data for convet tanscript ID</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"convertID_refseq_data.csv"</span>, package <span class="op">=</span> <span class="st">"MRAT"</span><span class="op">)</span>,
              stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>, encoding <span class="op">=</span> <span class="st">"UTF-8"</span>, row.names <span class="op">=</span> <span class="cn">NULL</span>, sep <span class="op">=</span> <span class="st">","</span><span class="op">)</span>
<span class="va">dat</span></code></pre></div>
<pre><code>##      refseq Nucleotide_changes
## 1 NM_005359              A947G
## 2 NM_000059             C3109T
## 3 NM_000059             A4599C</code></pre>
<p>Excute in convert_transcriptID()</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">new_dat</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/convert_transcriptID.html">convert_transcriptID</a></span><span class="op">(</span><span class="va">dat</span>, <span class="va">db</span>, dat_filter <span class="op">=</span> <span class="st">"refseq_mrna"</span><span class="op">)</span></code></pre></div>
<p>Result from the convert_transcriptID()</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">convertID_result</span>, package <span class="op">=</span> <span class="st">"MRAT"</span><span class="op">)</span>
<span class="va">convertID_result</span></code></pre></div>
<pre><code>##      refseq Nucleotide_changes CDS_start_loc ref alt chr ensembl_transcript_id
## 1 NM_000059             C3109T          3109   C   T  13       ENST00000544455
## 2 NM_000059             A4599C          4599   A   C  13       ENST00000544455
## 3 NM_005359              A947G           947   A   G  18       ENST00000342988
##   hgnc_symbol    start      end width strand         exon_id exon_rank tx_start
## 1       BRCA2 32911601 32911601     1      + ENSE00000939168        11     3336
## 2       BRCA2 32913091 32913091     1      + ENSE00000939168        11     4826
## 3       SMAD4 48586278 48586278     1      + ENSE00003494728         8     1485
##   tx_end   refsnp_id
## 1   3336  rs80358557
## 2   4826  rs80358694
## 3   1485 rs377119288</code></pre>
<p>Load the data for getting population allele frequency</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">anno_freq_data</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,
                          <span class="st">"anno_freq_data.csv"</span>,
                          package <span class="op">=</span> <span class="st">"MRAT"</span><span class="op">)</span>,
              stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>, encoding <span class="op">=</span> <span class="st">"UTF-8"</span>, row.names <span class="op">=</span> <span class="cn">NULL</span>, sep <span class="op">=</span> <span class="st">","</span><span class="op">)</span>
<span class="va">anno_freq_data</span></code></pre></div>
<pre><code>##   chr    start      end ref alt
## 1  18 48586278 48586278   A   G
## 2  13 32911601 32911601   C   T
## 3  13 32913091 32913091   A   C</code></pre>
<p>Excute in pop_freq()</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pop_dat</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/pop_freq.html">pop_freq</a></span><span class="op">(</span><span class="va">anno_freq_data</span>, pop<span class="op">=</span><span class="st">"db_gnomAD_exome_freq"</span><span class="op">)</span></code></pre></div>
<p>Result from the pop_freq()</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">pop_result</span>, package <span class="op">=</span> <span class="st">"MRAT"</span><span class="op">)</span>
<span class="va">pop_result</span></code></pre></div>
<pre><code>##   Chr    Start      End Ref Alt Func.knownGene Gene.knownGene
## 1  18 48586278 48586278   A   G         exonic          SMAD4
## 2  13 32911601 32911601   C   T         exonic          BRCA2
## 3  13 32913091 32913091   A   C         exonic          BRCA2
##   GeneDetail.knownGene ExonicFunc.knownGene
## 1                    .    nonsynonymous SNV
## 2                    .             stopgain
## 3                    .    nonsynonymous SNV
##                                                              AAChange.knownGene
## 1 SMAD4:uc002lfb.4:exon4:c.A482G:p.N161S,SMAD4:uc010xdp.2:exon8:c.A947G:p.N316S
## 2                                     BRCA2:uc001uub.1:exon11:c.C3109T:p.Q1037X
## 3                                     BRCA2:uc001uub.1:exon11:c.A4599C:p.K1533N
##   gnomAD_exome_ALL_Ref_freq gnomAD_exome_ALL_Alt_freq gnomAD_exome_AFR_Ref_freq
## 1                    0.9999                     1e-04                         1
## 2                    1.0000                     0e+00                         1
## 3                    0.9999                     1e-04                         1
##   gnomAD_exome_AFR_Alt_freq gnomAD_exome_AMR_Ref_freq gnomAD_exome_AMR_Alt_freq
## 1                         0                         1                         0
## 2                         0                         1                         0
## 3                         0                         1                         0
##   gnomAD_exome_ASJ_Ref_freq gnomAD_exome_ASJ_Alt_freq gnomAD_exome_EAS_Ref_freq
## 1                         1                         0                    0.9999
## 2                         1                         0                    0.9999
## 3                         1                         0                    0.9987
##   gnomAD_exome_EAS_Alt_freq gnomAD_exome_FIN_Ref_freq gnomAD_exome_FIN_Alt_freq
## 1                    0.0001                         1                         0
## 2                    0.0001                         1                         0
## 3                    0.0013                         1                         0
##   gnomAD_exome_NFE_Ref_freq gnomAD_exome_NFE_Alt_freq gnomAD_exome_SAS_Ref_freq
## 1                         1                         0                    0.9997
## 2                         1                         0                    1.0000
## 3                         1                         0                    1.0000
##   gnomAD_exome_SAS_Alt_freq gnomAD_exome_OTH_Ref_freq gnomAD_exome_OTH_Alt_freq
## 1                     3e-04                         1                         0
## 2                     0e+00                         1                         0
## 3                     0e+00                         1                         0</code></pre>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Ching-Yu Shih.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
